relay/pkg/header/integration_test.go

```diff
@@ -0,0 +1,292 @@
+package header
+
+import (
+	"context"
+	"math/big"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/keep-network/tbtc/relay/pkg/btc"
+	chainlocal "github.com/keep-network/tbtc/relay/pkg/chain/local"
+)
+
+func TestRelay_Integration(t *testing.T) {
+	ctx, cancelCtx := context.WithCancel(context.Background())
+	defer cancelCtx()
+
+	bc, err := btc.ConnectLocal()
+	if err != nil {
+		t.Fatal(err)
+	}
+	btcChain := bc.(*btc.LocalChain)
+
+	lc, err := chainlocal.Connect()
+	if err != nil {
+		t.Fatal(err)
+	}
+	localChain := lc.(*chainlocal.Chain)
+
+	headers := []*btc.Header{
+		{Hash: to32Bytes(0), Height: 0, PrevHash: to32Bytes(0), Raw: toBytes(0)},
+		{Hash: to32Bytes(1), Height: 1, PrevHash: to32Bytes(0), Raw: toBytes(1)},
+		{Hash: to32Bytes(2), Height: 2, PrevHash: to32Bytes(1), Raw: toBytes(2)},
+		{Hash: to32Bytes(3), Height: 3, PrevHash: to32Bytes(2), Raw: toBytes(3)},
+		{Hash: to32Bytes(4), Height: 4, PrevHash: to32Bytes(3), Raw: toBytes(4)},
+		{Hash: to32Bytes(5), Height: 5, PrevHash: to32Bytes(4), Raw: toBytes(5)},
+		{Hash: to32Bytes(6), Height: 6, PrevHash: to32Bytes(5), Raw: toBytes(6)},
+		{Hash: to32Bytes(7), Height: 7, PrevHash: to32Bytes(6), Raw: toBytes(7)},
+		// The first batch: difficulty change at the first header of the batch
+		{Hash: to32Bytes(8), Height: 8, PrevHash: to32Bytes(7), Raw: toBytes(8)},
+		{Hash: to32Bytes(9), Height: 9, PrevHash: to32Bytes(8), Raw: toBytes(9)},
+		{Hash: to32Bytes(10), Height: 10, PrevHash: to32Bytes(9), Raw: toBytes(10)},
+		{Hash: to32Bytes(11), Height: 11, PrevHash: to32Bytes(10), Raw: toBytes(11)},
+		{Hash: to32Bytes(12), Height: 12, PrevHash: to32Bytes(11), Raw: toBytes(12)},
+		// The second batch: headers span difficulty change (at header 16)
+		{Hash: to32Bytes(13), Height: 13, PrevHash: to32Bytes(12), Raw: toBytes(13)},
+		{Hash: to32Bytes(14), Height: 14, PrevHash: to32Bytes(13), Raw: toBytes(14)},
+		{Hash: to32Bytes(15), Height: 15, PrevHash: to32Bytes(14), Raw: toBytes(15)},
+		{Hash: to32Bytes(16), Height: 16, PrevHash: to32Bytes(15), Raw: toBytes(16)},
+		{Hash: to32Bytes(17), Height: 17, PrevHash: to32Bytes(16), Raw: toBytes(17)},
+		// The third batch: no difficulty change accross headers
+		{Hash: to32Bytes(18), Height: 18, PrevHash: to32Bytes(17), Raw: toBytes(18)},
+		{Hash: to32Bytes(19), Height: 19, PrevHash: to32Bytes(18), Raw: toBytes(19)},
+		{Hash: to32Bytes(20), Height: 20, PrevHash: to32Bytes(19), Raw: toBytes(20)},
+	}
+
+	btcChain.SetHeaders(headers)
+
+	// Set the best known digest to header 7, so the relay will be pulling headers
+	// starting with the 8th header
+	localChain.SetBestKnownDigest(to32Bytes(7))
+
+	// Start the relay with difficulty changing at every eighth header and
+	// a shortened pushing loop sleep time
+	const (
+		testDifficultyEpochDuration = 8
+		testRelayPushingSleepTime   = 200 * time.Millisecond
+	)
```

Comment:
Can we put those consts at the top of the file and document them? This way when someone reads this file he will be aware of them from the beginning.